src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -324,7 +324,17 @@ public void focusGained(final FocusEvent e) {
     m_rightHandSidePanel.add(m_smallView, BorderLayout.NORTH);
     m_tabsPanel.setBorder(null);
     m_rightHandSidePanel.add(m_tabsPanel, BorderLayout.CENTER);
-    m_actionButtons = new ActionButtons(m_data, m_mapPanel, this);
+
+    MovePanel movePanel = new MovePanel(m_data, m_mapPanel, this);
+    m_actionButtons = new ActionButtons(m_data, m_mapPanel, movePanel, this);
+
+    List<KeyListener> keyListeners = ImmutableList.of(this.getArrowKeyListener(), movePanel.getUndoMoveKeyListener());
+    for( KeyListener keyListener : keyListeners ) {
+      m_mapPanel.addKeyListener(keyListener );
+      // TODO: figure out if it is really needed to double add the key listener to both the frame and also the map panel
+      this.addKeyListener(keyListener);
```

Comment:
Bottom line, I don't want to make this PR depend on #305, which is a bug that existed in the system since I'm modelling a new feature after an existing one. Since the feature is also labelled as beta, I think it's reasonable to let some time go by while we try to figure out why we are getting #305